File: Dev/unitTest_Engine_cpp_gtest/Graphics/2D/EffectObject2D.cpp

#include "../../EngineTest.h"
#include <Altseed.h>
#include <gtest/gtest.h>

using namespace std;
using namespace asd;

class Graphics_EffectObject2D : public EngineTest
{
public:
	Graphics_EffectObject2D(bool isOpenGLMode) : EngineTest(u"EffectObject2D", isOpenGLMode, 20) {}

private:
	shared_ptr<EffectObject2D> effectObj;

protected:
	void OnStart()
	{
		auto scene = make_shared<Scene>();
		auto layer = make_shared<Layer2D>();
		auto effect = asd::Engine::GetGraphics()->CreateEffect(u"Data/Effect/Laser01.efk");
		effectObj = make_shared<EffectObject2D>();

		layer->AddObject(effectObj);
		scene->AddLayer(layer);
		asd::Engine::ChangeScene(scene);

		effectObj->SetPosition(asd::Vector2DF(320, 240));
		effectObj->SetScale(asd::Vector2DF(20, 20));
		effectObj->SetEffectRotation(90);

		effectObj->SetEffect(effect);
	}

	void OnUpdating()
	{
		if (m_currentTime == 1)
		{
			effectObj->Play();
		}
	}
};

ENGINE_TEST(Graphics, EffectObject2D)

File: Dev/unitTest_Engine_cs/Graphics/2D/EffectObject2D.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace unitTest_Engine_cs.Graphics._2D
{
	class EffectObject2D : EngineTest
	{
		asd.EffectObject2D m_effectObj;

		public EffectObject2D()
			: base(20)
		{
		}

		protected override void OnStart()
		{
			var scene = new asd.Scene();
			var layer = new asd.Layer2D();
			var effect = asd.Engine.Graphics.CreateEffect("Data/Effect/Laser01.efk");
			m_effectObj = new asd.EffectObject2D();

			layer.AddObject(m_effectObj);
			scene.AddLayer(layer);
			asd.Engine.ChangeScene(scene);

			m_effectObj.Position = new asd.Vector2DF(320, 240);
			m_effectObj.Scale = new asd.Vector2DF(20, 20);
			m_effectObj.EffectRotation = 90;

			m_effectObj.Effect = effect;
			m_effectObj.Play();
		}

		protected override void OnUpdating()
		{
			base.OnUpdating();
		}
	}
}
